### ISSUE TITLE
Can't parse macros with semicolons in their arguments

### ISSUE THREAD

**Issue by thezeroalpha** [newsboat/newsboat#1200]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: 2.21.0

Previously, I had this macro in my config file, to save a selected article to Pocket:

```
macro s set browser "{ open -gj 'pocket://add?url=%u' && sleep 3 && killall Pocket; } 1>/dev/null 2>&1 & disown"; open-in-browser ; set browser "open -g '%u'"
```

The aim was to group the three commands (`open`, `sleep`, `killall`) using braces, and background/disown them together, so that I could continue using Newsboat while the article was being saved.

However, since the new update (presumably because of the point "It is now a startup error for a macro to have no operations"), I get the following error while starting Newsboat:

> Starting newsboat 2.21.0...
> Loading configuration...Error while processing command \`macro s set browser "{ open -gj 'pocket://add?url=%u' && sleep 3 && killall Pocket; } 1>/dev/null 2>&1 & disown"; open-in-browser ; set browser "open -g '%u'"' (~/.config/newsboat/config line 59): \`}' is not a valid operation

Is this intended? Also, is there currently any way to achieve what I'm trying to do (i.e. group multiple shell commands together, redirect their streams, and disown them together) without getting a start-up error?

Edit: I understand it's also possible to do it using an external script and then calling the script from the macro, but I'd prefer if it were possible to avoid external scripts. My question is more regarding whether this is intended as a change that breaks command grouping, or if it was just missed when detecting operations in macros.

**Comment by Minoru** [comment]
Hi! This definitely is *not* intended.

Bisect points at 58096742a43755f66257f48a40d2e71f778caca4 which is part of #1119. I was a reviewer for that one, and failed to consider that operations might have arguments that contain semicolons.

Moving the whole thing into a separate script is the only workaround that I can suggest.

Also note that `%u` is replaced by URL enclosed in single quotes, so the `'pocket://add?url=%u'` thing expands to something like `'pocket://add?url='https://example.com/test post''`

**Comment by thezeroalpha** [comment]
Ah alright, so it's because of the semicolons. That makes sense, thanks for the info!

**Comment by Minoru** [comment]
@thezeroalpha I just submitted #1203 which fixes this. If you're able to build from Git, please do so and test. Build instructions are in the README.

I'm haven't yet made up my mind about making a patch release for this. Looks like you're the only one who's affected. @uzhne, did you run into this bug as well?

**Comment by thezeroalpha** [comment]
@Minoru thanks for the fix. Unfortunately I'm not able to build from Git to test it, I'm on macOS where building and getting the dependencies right isn't trivial, and I can't get Docker working after spending three hours on it.

**Comment by piperfw** [comment]
I also have this issue with commands containing semicolons (that previously worked).

**Comment by Minoru** [comment]
@thezeroalpha You can follow [what we do on CI servers](https://github.com/newsboat/newsboat/blob/cb4a280b1badadc70893bb5cedd9cbd3d0f91a36/.cirrus.yml#L99-L114) to install all the dependencies. Homebrew makes it easy.

Regarding Docker: is the problem with Docker itself, or is it with our Dockerfiles? We don't use Docker on macOS, so no guarantees there, but if you did discover some problems with our files, it's be nice to fix them.

I'm not sure if you have time and inclination to dig into any of that. Feel free to ignore the above suggestions.

@piperfw Can you post your macros? I'll include them as tests, to get more assurance that my fix works.

Also: thanks for commenting. I'm now more inclined to do a patch release.

Meanwhile, a review of #1203 uncovered that my fix introduces other regressions. I'm now re-thinking the approach, and will develop a different fix. I'll comment here when that's done.

**Comment by piperfw** [comment]
@Minoru certainly, thank you for continuing to work on this issue and of course the fantastic newsboat.

The macro
```
macro p set browser "echo %u >> log.txt; mpv %u"; open-in-browser-and-mark-read ; set browser "chromium %u"
```
(`echo %u` was previously a local logging script, but the error is the same)

Leads to the following 

```
Starting newsboat 2.21.0...
Loading configuration...Error while processing command `macro p set browser "echo %u >> log.txt; mpv %u"; open-in-browser-and-mark-read ; set browser "chromium %u"' (/home/pip/.newsboat/config line 18): `mpv' is not a valid operation
```

and newsboat fails to launch.

**Comment by thezeroalpha** [comment]
@Minoru nice thanks, in that case I'll see if I can test it. 
As for Docker, I'm not sure if it was an issue with the image or with Docker itself on Mac (since it also has to interplay with Virtualbox via docker-machine). I forget the exact error message, but I was getting input/output errors while trying to provision the container (installing dependencies etc.). It wasn't due to lack of disk space, and restarting/rerunning/recreating didn't help. But it doesn't seem like a problem with the Dockerfiles.

**Comment by thezeroalpha** [comment]
I've tested #1203 on macOS version 10.14.6, the macros now work as expected. Thank you @Minoru!
From my side, this issue can be closed now, but I'm not closing it myself in case you'd like to leave it open until the PR is merged.

**Comment by Minoru** [comment]
@thezeroalpha Thanks for checking! Turns out that PR introduces regressions of its own, so I'm reworking it. So maybe don't dismantle your build environment just yet; it's be nice if you could check the updated PR as well.

**Comment by Minoru** [comment]
@thezeroalpha Sorry it took me so long to get out a new version of the PR — it got blocked on internal changes that turned out to be *much* harder than I anticipated. But I just pushed a new version, so if you haven't torn down the build setup yet, it'd be nice if you could test that one as well. Thanks for your patience!

**Comment by thezeroalpha** [comment]
@Minoru Totally understandable, no worries. I've checked the version merged as part of #1203, everything works fine for me. Thank you for your work on this!
